docs/swift-3-api.diff

```diff
@@ -0,0 +1,399 @@
++    public var _nsError: NSError
++    public init(_nsError error: NSError)
++public typealias _ErrorType = RLMError
++    public static var _nsErrorDomain = RLMErrorDomain
+ public class LinkingObjectsBase: NSObject, NSFastEnumeration
+-    public func countByEnumeratingWithState(state: UnsafeMutablePointer<NSFastEnumerationState>, objects buffer: AutoreleasingUnsafeMutablePointer<AnyObject?>, count len: Int) -> Int
++    public func countByEnumerating(with state: UnsafeMutablePointer<NSFastEnumerationState>, objects buffer: AutoreleasingUnsafeMutablePointer<AnyObject?>!, count len: Int) -> Int
+ public final class LinkingObjects<T: Object>: LinkingObjectsBase
+     public subscript(index: Int) -> T
+     public typealias Element = T
+     public var realm: Realm? 
+-    public var invalidated: Bool 
++    public var isInvalidated: Bool
+     public var count: Int 
+     public init(fromType type: T.Type, property propertyName: String)
+     public override var description: String
+-    public func indexOf(object: T) -> Int?
+-    public func indexOf(predicate: NSPredicate) -> Int?
+-    public func indexOf(predicateFormat: String, _ args: AnyObject...) -> Int?
++    public func index(of object: T) -> Int?
++    public func indexOfObject(for predicate: NSPredicate) -> Int?
++    public func indexOfObject(for predicateFormat: String, _ args: Any...) -> Int?
+     public var first: T? 
+     public var last: T? 
+-    public override func valueForKey(key: String) -> AnyObject?
+-    public override func valueForKeyPath(keyPath: String) -> AnyObject?
+-    public override func setValue(value: AnyObject?, forKey key: String)
+-    public func filter(predicateFormat: String, _ args: AnyObject...) -> Results<T>
+-    public func filter(predicate: NSPredicate) -> Results<T>
+-    public func sorted(property: String, ascending: Bool = true) -> Results<T>
+-    public func sorted<S: SequenceType where S.Generator.Element == SortDescriptor>(sortDescriptors: S) -> Results<T>
+-    public func min<U: MinMaxType>(property: String) -> U?
+-    public func max<U: MinMaxType>(property: String) -> U?
+-    public func sum<U: AddableType>(property: String) -> U
+-    public func average<U: AddableType>(property: String) -> U?
+-    public func addNotificationBlock(block: (RealmCollectionChange<LinkingObjects> -> Void)) -> NotificationToken
+-    public func generate() -> RLMGenerator<T>
++    public override func value(forKey key: String) -> Any?
++    public override func value(forKeyPath keyPath: String) -> Any?
++    public override func setValue(_ value: Any?, forKey key: String)
++    public func filter(using predicateFormat: String, _ args: Any...) -> Results<T>
++    public func filter(using predicate: NSPredicate) -> Results<T>
++    public func sorted(onProperty property: String, ascending: Bool = true) -> Results<T>
++    public func sorted<S: Sequence>(with sortDescriptors: S) -> Results<T> where S.Iterator.Element == SortDescriptor
```

Comment:
`onProperty` vs `with` feels somewhat inconsistent. I can see why `with` was used as sorting _on_ a sort descriptor doesn't seem grammatically correct. Perhaps this should be "sorted _by_ property" / "sorted _by_ sort descriptors" instead, with `byProperty` and `by` being the labels?